notebooks/contributing/formatting.md

```diff
@@ -0,0 +1,289 @@
+observation locations, and a list of field components
+which are to be measured. For the tutorial simulation, the receivers are located
+5 m above the surface topography and spaced 10 m apart.
+```
+
+If something has been thoroughly explained in an introductory tutorial, you can avoid repetition by linking to that tutorial.
+However, the choice in values used in your tutorial should always be stated. E.g.
+
+```
+## Assign Uncertainties
+
+Approaches for applying reasonable uncertainties to normalized voltage and apparent
+resistivity data were presented in the [2.5D Inversion](inv_dcr_2d.ipynb) tutorial.
+Here, we apply uncertainties of 1e-7 V/A + 10 % to the normalized voltage data being
+inverted.
+```
+
+### Code Block Cells
+
+Code blocks must follow the [style required by SimPEG](https://docs.simpeg.xyz/latest/content/getting_started/contributing/code-style.html).
+Please provide comments when defining new objects. This is especially important when exposing the reader to new functionality.
+Avoid lengthy code blocks that can be better parsed into multiple cells. Markdown cells can also be added between code cells
+to provide additional explanation to the reader.
+
+```python
+# Define the component(s) of the field we want to simulate as strings within
+# a list. Here we simulate only the vertical component of the gravity anomaly.
+components = ["gz"]
+
+# Use the observation locations and components to define receivers for the entire survey
+# in one step. The set of receivers, even if it's only 1, are organized within a list.
+receiver_list = gravity.receivers.Point(receiver_locations, components=components)
+receiver_list = [receiver_list]
+
+# Defining the source. For gravity surveys, we simply need to specific the list of
+# receivers associated with the source field.
+source_field = gravity.sources.SourceField(receiver_list=receiver_list)
+
+# Defining the survey.
+survey = gravity.survey.Survey(source_field)
+```
+
+### Plotting Requirements
+
+The contributor is required to plot data, models and other quantities that are best understood visually.
+Due to its widespread use, figures should be generated using [matplotlib](https://matplotlib.org/) package.
+When generating a figure, please ensure:
+
+* the figure renders at an appropriate size when building the website
+* the code used to generate the figure is compact whenever possible
+* the text and features within the plot are legible
+
+
+## Check Style of Notebooks
+
+We can check and autoformat the code style within the notebooks using [`ruff`][ruff] and
+[`nbqa`][nbqa].
+
+To check the style of the notebooks:
+
+```bash
+nbqa ruff notebooks
+```
+
+To perform auto reformatting on the notebooks:
+
+```bash
+nbqa ruff --fix notebooks
+```
+
+Alternatively, you can use the targets we have in the `Makefile`, like `make
+check` and `make format`. Read more information about the available targets
+by running `make help`.
```

Comment:
I updated these instructions in https://github.com/simpeg/user-tutorials/pull/51. This is just a note to myself to remind updating them here as well.

Reply:
Done!